src/test/java/teammates/test/cases/util/ActivityLogGeneratorTest.java

```diff
@@ -0,0 +1,262 @@
+package teammates.test.cases.util;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import com.google.appengine.api.log.AppLogLine;
+
+import teammates.common.datatransfer.AccountAttributes;
+import teammates.common.datatransfer.StudentAttributes;
+import teammates.common.datatransfer.UserType;
+import teammates.common.exception.PageNotFoundException;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.ActivityLogEntry;
+import teammates.common.util.ActivityLogGenerator;
+import teammates.common.util.Const;
+import teammates.common.util.EmailWrapper;
+import teammates.logic.api.EmailGenerator;
+import teammates.test.cases.BaseTestCase;
+import teammates.test.driver.AssertHelper;
+import teammates.test.driver.GaeSimulation;
+
+/**
+ * SUT: {@link ActivityLogGenerator}
+ */
+public class ActivityLogGeneratorTest extends BaseTestCase {
+    
+    private ActivityLogGenerator logCenter = new ActivityLogGenerator();
+    
+    private GaeSimulation gaeSimulation = GaeSimulation.inst();
```

Comment:
The fact that you need `GaeSimulation` tells at least one thing not done right here:
- This test should inherit `BaseComponentTestCase`, and thus the component should be under `logic` or `storage`.
- This simulation should not be necessary and all variables generated from the simulation can be hardcoded.